src/GEXF/metagraphs.jl

```diff
@@ -0,0 +1,91 @@
+using MetaGraphs
+
+function idandtypes(prs)
+    k2id = Dict{Symbol,Int}()
+    k2type = Dict{Symbol,Type}()
+    i = 1
+    for vs in values(prs)
+        for (k,v) in (vs)
+            if haskey(k2id, k)
+                if k2type[k] != typeof(v)
+                    k2type[k] = promote_type(k2type[k],typeof(v))
+                end
+            else
+                k2id[k] = i 
+                i += 1
+                k2type[k] = typeof(v)
+            end 
+        end
+    end
+    return(k2id, k2type)
+end
+
+type2string(::Type{T}) where {T<:Integer} = "integer"
+type2string(::Type{T}) where {T<:AbstractFloat} = "double"
+type2string(::Type{T}) where {T<:Char} = "string"
+type2string(::Type{T}) where {T<:AbstractString} = "string"
+
+function exportattributes(xg, prs, element_type)
+    (k2id, k2type) = idandtypes(prs)
+    if !isempty(k2id)
+        xattrs = addelement!(xg, "attributes")
+        xattrs["class"] = element_type
+        for (k,i) in k2id
+            xat=addelement!(xattrs, "attribute")
+            xat["id"] = i
+            xat["title"] = k
+            xat["type"] = type2string(k2type[k])
+        end
+    end
+    k2id
+end
+
+function add_properties!(xv, ks, prs)
+    isempty(prs) && return
+    av = addelement!(xv, "attvalues")
+    for (k,v) in prs
+        av = addelement!(av, "attvalue")
+        av["for"] = ks[k]
+        av["value"] = v
+    end
+end
+
+function savegexf(io::IO, g::MetaGraph, gname::String)
```

Comment:
Should this be `MetaGraph` or `AbstractMetaGraph`? Is it just for undirected metagraphs?